pages/p/[productId].js

```diff
@@ -121,6 +133,25 @@ const Product = React.memo(lazyProps => {
     </Row>
   )
 
+  // Fetch variant data upon changing color or size options
+  useDidMountEffect(() => {
+    const url = URL.format({
+      pathname: `/api/p/${product.id}`,
+      query: {
+        color: get(color, 'id'),
+        size: get(size, 'id'),
+      },
+    })
+    fetch(url)
+      .then(res => res.json())
+      .then(data => {
+        updateState({
+          ...state,
+          pageData: { ...state.pageData, ...data },
+        })
+      })
+  }, [get(color, 'id'), get(size, 'id')])
```

Comment:
After implementing the get() || {} we can just use everywhere now color.id and so on, should not crash.